### DISCUSSION TITLE
I want to automatically execute some commands when I open a new tab

### DISCUSSION THREAD

**Discussion by xiaoxuuu** [Eugeny/tabby#7425] [General]
Like executing command `source ~/.bash_profile` automatically when opening local tags.

I've tried the plugin "quick commands" but it needs to be done manually. I want to open a new tab automatically.

I would like to be able to extend the "command line" capabilities of "Profiles&connections" so that custom commands can be written.

**Comment by Eugeny** [comment]
You're looking for the "Login scripts" tab in the profile settings.

**Reply by caseyredisatwork** [reply]
I'd like to do the opposite of what the "Login scripts" functionality provides - I have a machine that I ssh into that is setup to sleep after 10 minutes and what I'd like to do (and have setup in Moba XTerm which I'm trying to migrate from) is execute a WOL tool and wait a few seconds before trying to connect. Is there anything I can do in Tabby along those lines?

**Reply by xiaoxuuu** [reply]
> You're looking for the "Login scripts" tab in the profile settings.

Yes, I saw it in SSH connection, which is very useful, but I want to use it on bash.

I am a Mac user. There is some information in `~/.bash_profile`.

So I hope Tabby can automatically execute some commands when opening the bash window.

**Reply by Eugeny** [reply]
Tabby just launches your shell, which will automatically read bash_profile/bashrc/any alternative files it might be using.

**Reply by caseyredisatwork** [reply]
Is there any way to run a command prior to making the ssh connection?
